pkg/reconciler/v1alpha1/taskrun/config/store.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2018 The Knative Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"context"
+
+	"github.com/knative/pkg/configmap"
+)
+
+type cfgKey struct{}
+
+// +k8s:deepcopy-gen=false
+type Config struct {
+	Entrypoint *Entrypoint
+}
+
+func FromContext(ctx context.Context) *Config {
+	return ctx.Value(cfgKey{}).(*Config)
+}
+
+func ToContext(ctx context.Context, c *Config) context.Context {
+	return context.WithValue(ctx, cfgKey{}, c)
+}
+
+// +k8s:deepcopy-gen=false
+type Store struct {
+	*configmap.UntypedStore
+}
+
+func NewStore(logger configmap.Logger) *Store {
```

Comment:
can you tell us more about why we need the `store.go`, it seems like it would be simpler without it similar to how the logging config is updated in the main controller cmd

https://github.com/knative/build-pipeline/blob/1096064186fa32d0a5816a0fd1a65ec1ff757958/cmd/controller/main.go#L141

Reply:
To be honest, i am a rookie in Knative, so i referred to the implementation of [Knative/serving](https://github.com/knative/serving/tree/master/pkg/reconciler/v1alpha1/revision/config).
I just feel this approach has good scalability and readability.